### DISCUSSION TITLE
Undo/reapply transactions

### DISCUSSION THREAD

**Discussion by somelinguist** [fsprojects/FSharp.Data.Adaptive#111] [Q&A] [answered]
I’m currently trying to rewrite a project to use the Adaptive library. I got to the part where my model interacts with my undo/redo system. I’m wondering if there is a way I could take advantage of the library’s incremental nature to role back or reapply transactions manually as part of the system? Sorry if I’m missing something obvious. :) Thanks!

**Comment by krauthaufen** [comment, marked as answer]
well if you're using adaptify you can just update the `AdaptiveModel` to a stored immutable snapshot right? (this will result in minimal output changes)

We currently have nothing to memorize changes internally since we use adaptify and immutable models on the input-side where undo/redo is relatively trivial.

**Reply by somelinguist** [reply]
Thanks! That’s how the current undo system works, so that’s easy. I just wanted to make sure I wasn’t missing something better. :)
